README.md
# Airship 

> Airship is a components library built by CARTO

[![CircleCI](https://circleci.com/gh/CartoDB/airship/tree/master.svg?style=svg)](https://circleci.com/gh/CartoDB/airship/tree/master)

| module | stable | description |
|---|---|---|
|  [airship-style](https://www.npmjs.com/package/@carto/airship-style) | ![npm version](https://badgen.net/npm/v/@carto/airship-style) | A complete CSS framework to build location intelligence apps. |
|  [airship-components](https://www.npmjs.com/package/@carto/airship-components) | ![npm version](https://badgen.net/npm/v/@carto/airship-components) | Web components for Location Intelligence apps. |
|  [airship-icons](https://www.npmjs.com/package/@carto/airship-icons) | ![npm version](https://badgen.net/npm/v/@carto/airship-icons) | High quality icons set |

## What is Airship?

Airship is a set of tools designed to facilitate the development of Location Intelligence apps by offering layouts, basic patterns, templates, CSS classes, components, widgets and much more. Apps created with Airship will have a consistent UI and an proper UX validated and tested by CARTO.

Airship is currently composed of three fundamental parts: A [CSS framework](https://carto.com/developers/airship/guides/styles/), a library of [Web Components](https://carto.com/developers/airship/guides/web-components/) and a set of [icons](https://carto.com/developers/airship/guides/icons/).

In a Location Intelligence application you can distinguish two parts: the map itself and everything else. Airship helps you with `everything else`, being responsible for the layout and UI of the application. To render the map, you can use one of our rendering libraries like [CARTO.js](https://github.com/CartoDB/carto.js) or [CARTO-VL](https://github.com/CartoDB/carto-vl).

## Documentation

Check out our [documentation website](https://carto.com/developers/airship/)

## Integration with frameworks

Using Airship within your workflow is easy.

For simple elements, load [Airship styles](https://carto.com/developers/airship/guides/styles/) and use its classes in your application along with your own CSS if needed.

For widgets, as Airship widgets are Web Components, they can be integrated in your framework, whether is [React](https://carto.com/developers/airship/guides/integrating-react/), [Vue](https://carto.com/developers/airship/guides/integrating-Vue/), [Angular](https://carto.com/developers/airship/guides/integrating-Angular/) or the newest kid on the block. Read our guides to know how to use them in your current flow.

## License
BSD-3-Clause, see the included [LICENSE.md](LICENSE.md) file.


CHANGELOG.md
# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [2.4.1](https://github.com/CartoDB/airship/compare/v2.4.0...v2.4.1) (2020-08-31)

### Features

* **components:** add option `show-clear-button` for histogram and time series widgets ([#707](https://github.com/CartoDB/airship/pulls/707)) ([fe41775a](https://github.com/CartoDB/airship/commit/fe41775a9c7470efeae1784de0c629da0450eb3e))

* **components:** add width & height parameters to Infowindow component ([#708](https://github.com/CartoDB/airship/pulls/708)) ([9fe287d7](https://github.com/CartoDB/airship/commit/9fe287d72f10c9b6150ad85de2d71e507275745d))


# [2.4.0](https://github.com/CartoDB/airship/compare/v2.4.0-rc.0...v2.4.0) (2020-07-03)


### Bug Fixes

* **styles:** Add overflow hidden to #map ([#698](https://github.com/CartoDB/airship/issues/698)) ([f4abb11](https://github.com/CartoDB/airship/commit/f4abb11a938e5ac7f92bfd99e1278ce957fc8cdb))



# [2.4.0-rc.0](https://github.com/CartoDB/airship/compare/v2.3.4...v2.4.0-rc.0) (2020-06-24)


### Features

* **components:** Formula Widget ([#696](https://github.com/CartoDB/airship/issues/696)) ([edea7aa](https://github.com/CartoDB/airship/commit/edea7aa548a56ba7aa1215f90b17ac72e9c0ffe8))



## [2.3.4](https://github.com/CartoDB/airship/compare/v2.3.3...v2.3.4) (2020-05-29)


### Bug Fixes

* Return to SI format for values using precision higher than 3 and values higher than 999 ([#695](https://github.com/CartoDB/airship/issues/695)) ([392b960](https://github.com/CartoDB/airship/commit/392b960b90209659c93627a0632627d3533edafb))



## [2.3.3](https://github.com/CartoDB/airship/compare/v2.3.2...v2.3.3) (2020-05-22)


### Bug Fixes

* Apply SI format for histogram axes only if domain difference is lower than 0.001 ([52d8d08](https://github.com/CartoDB/airship/commit/52d8d080c3831e94840f8ea2625944cd96730d2b))
* set new rules to format X axis for histogram [ch76214] ([2933c71](https://github.com/CartoDB/airship/commit/2933c71d1a31899233cdecd0aee4a0e7ae8007d7))



## [2.3.2](https://github.com/CartoDB/airship/compare/v2.3.2-rc.2...v2.3.2) (2020-05-13)



## [2.3.2-rc.2](https://github.com/CartoDB/airship/compare/v2.3.2-rc.0...v2.3.2-rc.2) (2020-05-08)



## [2.3.2-rc.0](https://github.com/CartoDB/airship/compare/v2.3.1...v2.3.2-rc.0) (2020-04-30)



## [2.3.1](https://github.com/CartoDB/airship/compare/v2.3.1-rc.2...v2.3.1) (2020-04-03)


### Bug Fixes

* manual reset back packages version before bump in new release ([ce71061](https://github.com/CartoDB/airship/commit/ce710614facaf82387e06e7da4027f189ec1e86c))



## [2.3.1-rc.2](https://github.com/CartoDB/airship/compare/v2.3.1-rc.1...v2.3.1-rc.2) (2020-04-03)



## [2.3.1-rc.1](https://github.com/CartoDB/airship/compare/v2.3.1-rc.0...v2.3.1-rc.1) (2020-04-02)


### Bug Fixes

* linter issues ([eb8f041](https://github.com/CartoDB/airship/commit/eb8f0419debe1713ecbdd66fa766487ebb72e690))
* tests not being executed due to describe with async ([0eb4659](https://github.com/CartoDB/airship/commit/0eb4659f570437f6470c3b7eb5da52d5a8a6f718))
* tslint error detecting an es2017 feature on bridge ([4f3035b](https://github.com/CartoDB/airship/commit/4f3035b6408434b53dc94fb87e01a3ea03028456))



## [2.3.1-rc.0](https://github.com/CartoDB/airship/compare/v2.3.0...v2.3.1-rc.0) (2020-03-18)



# [2.3.0](https://github.com/CartoDB/airship/compare/v2.2.1...v2.3.0) (2020-02-24)



## [2.2.1](https://github.com/CartoDB/airship/compare/v2.2.0...v2.2.1) (2020-01-21)


### Bug Fixes

* Developer docs ([099bd11](https://github.com/CartoDB/airship/commit/099bd11daa115a0202971ddf7be0d5bce9977703))
* **components:** Add loading placeholders for widgets ([57802e3](https://github.com/CartoDB/airship/commit/57802e353f429b5f39d09b83425d9a88adf5ee57))



# [2.2.0](https://github.com/CartoDB/airship/compare/v2.1.1...v2.2.0) (2020-01-17)


### Bug Fixes

* update dependencies to fix vulnerabilities ([13e7c90](https://github.com/CartoDB/airship/commit/13e7c90700457998d2052d84924ef745ff5eefeb))
* **components:** time series / histogram improvements ([#641](https://github.com/CartoDB/airship/issues/641)) ([31789ae](https://github.com/CartoDB/airship/commit/31789ae0e1c253fa754a983be0d699254d2bb618))



# [2.2.0-rc.1](https://github.com/CartoDB/airship/compare/v2.2.0-rc.0...v2.2.0-rc.1) (2019-07-16)


### Bug Fixes

* **bridge:** Fix bridge tests ([1b2c1d7](https://github.com/CartoDB/airship/commit/1b2c1d74fd67feca48b6d34f73ca2ae82b89d9bd))


### Features

* **components:** Animation controls API ([#634](https://github.com/CartoDB/airship/issues/634)) ([8fc2d1a](https://github.com/CartoDB/airship/commit/8fc2d1af54629c282ec7e7185fd7f239aa06358c))
* **components:** Improve animation controls API ([#635](https://github.com/CartoDB/airship/issues/635)) ([5a24e3a](https://github.com/CartoDB/airship/commit/5a24e3a2bc6f3597ac4e18a734d0648daf7187c5))
* **components:** Time Series date format improvements ([#638](https://github.com/CartoDB/airship/issues/638)) ([ef6fec8](https://github.com/CartoDB/airship/commit/ef6fec8a49ba3f600ee26517314467764b57c8f4)), closes [CartoDB/cartoframes#813](https://github.com/CartoDB/cartoframes/issues/813)




src/App.js
import React, { Component } from 'react';
import {
  Grid,
  Jumbo,
  Display,
  Title,
  Subheader,
  Text,
  Caption,
  Button,
  ButtonGroup,
  PlusIcon,
  SearchIcon,
  PointIcon,
  LeftArrowIcon,
  RightArrowIcon,
  TopArrowIcon,
  BottomArrowIcon,
  ChevronDownIcon,
  ChevronUpIcon,
  ChevronLeftIcon,
  ChevronRightIcon,
  MenuIcon,
  CloseIcon,
  CheckIcon,
  PencilIcon,
  HomeIcon,
  InfoIcon,
  MinusIcon,
  PointsIcon,
  QuestionIcon,
  SettingsIcon,
  CheckFillIcon,
  CheckRoundedIcon,
  AlertIcon,
  AlertFillIcon,
  Tabs,
  Dropdown,
  Table,
  Toggle,
  Dropshadow,
  Range,
  Breadcrumb,
  Loading,
  Avatar,
  Badge,
  Tooltip,
  Collapsible,
  Steps,
  Legend,
  Flag,
  Banner,
  Datepicker,
  Radiobutton,
  Checkbox,
  Typeahead
} from './components';
import { colors } from './constants';

class App extends Component {
  render() {
    return (
      <div>
        <Grid width={300} gap={20} align="center">
          <div>Column</div>
          <div>Column</div>
          <div>Column</div>
          <div>Column</div>
          <div>Column</div>
        </Grid>
        <Jumbo margin="0 0 20px">
          I'm thinking two circus clowns dancing. You?
        </Jumbo>
        <Jumbo as="h2">I'm a Jumbo h2</Jumbo>
        <Jumbo as="h2" font="mono">
          I'm a Jumbo h2 mono
        </Jumbo>
        <Jumbo as="h2" weight="medium">
          I'm a Jumbo h2 medium
        </Jumbo>
        <Jumbo as="h3" color="#ff9900">
          I'm a Jumbo h3
        </Jumbo>
        <Display margin="0 0 20px">
          I'm thinking two circus clowns dancing. You?
        </Display>
        <Display as="h2">I'm a Display h2</Display>
        <Display as="h3" color="#ff9900">
          I'm a Display h3
        </Display>
        <Title>I'm thinking two circus clowns dancing. You?</Title>
        <Title as="h2">I'm a Title h2</Title>
        <Title as="h3" color="#ff9900">
          I'm a Title h3
        </Title>
        <Title as="h3" weight="medium">
          I'm a Title h3
        </Title>
        <Subheader>I'm thinking two circus clowns dancing. You?</Subheader>
        <Subheader as="h2">I'm a Subheader h2</Subheader>
        <Subheader as="h2" weight="medium">
          I'm a Subheader h2
        </Subheader>
        <Subheader as="h2" font="mono">
          I'm a Subheader h2 mono
        </Subheader>
        <Subheader as="h3" color="#ff9900">
          I'm a Subheader h3
        </Subheader>
        <Text>I'm thinking two circus clowns dancing. You?</Text>
        <Text as="div">I'm a Text div</Text>
        <Text color="#ff9900">I'm a Text colored</Text>
        <Text as="div" font="mono">
          I'm a Text div mono
        </Text>
        <Caption as="span">I'm a Caption span</Caption>
        <Caption font="mono" color="skyblue">
          I'm a Caption mono as span
        </Caption>
        <Caption as="div" weight="medium" color="skyblue">
          I'm a Caption medium
        </Caption>
        <div style={{ marginLeft: '10px' }}>
          <h3>Button text only</h3>
          <Button>Button</Button>
          <Button large>Button large</Button>
          <Button small>Button small</Button>
          <h3>Button icon + text</h3>
          <Button>
            <PlusIcon />
            Button
          </Button>
          <Button large>
            <PlusIcon />
            Button
          </Button>
          <Button small>
            <PlusIcon />
            Button
          </Button>
          <h3>Button icon only</h3>
          <Button>
            <PlusIcon />
          </Button>
          <Button large>
            <PlusIcon />
          </Button>
          <Button large>Large</Button>
          <h3>Button secondary</h3>
          <Button secondary>Button</Button>
          <Button secondary large>
            Button large
          </Button>
          <Button secondary large>
            <PlusIcon />
            Button large
          </Button>
          <Button secondary large disabled>
            <PlusIcon />
            Button large
          </Button>
          <Button secondary small>
            Button small
          </Button>
          <h3>Button borderless</h3>
          <Button borderless>Button</Button>
          <Button borderless large>
            Button large
          </Button>
          <Button borderless small>
            Button small
          </Button>
        </div>
        <div>
          <div style={{ padding: '10px' }}>
            <ButtonGroup>
              <Button>
                <PlusIcon />
              </Button>
              <Button>Mundo</Button>
              <Button>Happy</Button>
            </ButtonGroup>
          </div>
          <div style={{ padding: '10px' }}>
            <ButtonGroup secondary>
              <Button>
                <PlusIcon />
              </Button>
              <Button>Mundo</Button>
              <Button>Happy</Button>
            </ButtonGroup>
          </div>
        </div>
        <div>
          <PlusIcon /> <SearchIcon /> <PointIcon /> <LeftArrowIcon />{' '}
          <RightArrowIcon /> <TopArrowIcon /> <BottomArrowIcon />{' '}
          <ChevronDownIcon /> <ChevronUpIcon /> <ChevronLeftIcon />{' '}
          <ChevronRightIcon /> <MenuIcon /> <CloseIcon /> <CheckIcon />{' '}
          <PencilIcon /> <HomeIcon /> <InfoIcon /> <MinusIcon /> <PointsIcon />{' '}
          <QuestionIcon /> <SettingsIcon /> <CheckFillIcon />{' '}
          <CheckRoundedIcon /> <AlertIcon /> <AlertFillIcon />
        </div>
        <div>
          <Tabs selected={1}>
            <Tabs.Panel label="wadus">Hola</Tabs.Panel>
            <Tabs.Panel label="world">Mundo</Tabs.Panel>
          </Tabs>
        </div>
        <div>
          <Tabs selected="world">
            <Tabs.Panel label="wadus">Hola</Tabs.Panel>
            <Tabs.Panel label="world">Mundo</Tabs.Panel>
          </Tabs>
        </div>
        <div>
          <Tabs large>
            <Tabs.Panel label="wadus">Hola</Tabs.Panel>
            <Tabs.Panel label="world">Mundo</Tabs.Panel>
          </Tabs>
        </div>
        <div style={{ padding: '10px' }}>
          <Dropdown>
            <Dropdown.Trigger>
              <Button>Click me</Button>
            </Dropdown.Trigger>
            <Dropdown.Content>
              <Dropdown.Menu>
                <Dropdown.Item>All</Dropdown.Item>
                <Dropdown.Item>Open</Dropdown.Item>
                <Dropdown.Item>Fullfilled</Dropdown.Item>
                <Dropdown.Item>Close</Dropdown.Item>
              </Dropdown.Menu>
            </Dropdown.Content>
          </Dropdown>
        </div>
        <div style={{ padding: '10px' }}>
          <Dropdown action="over">
            <Dropdown.Trigger as="span">
              <Button borderless>
                Hover me&nbsp;<ChevronDownIcon width={8} height={4} />
              </Button>
            </Dropdown.Trigger>
            <Dropdown.Content>
              <Dropdown.Menu>
                <Dropdown.Item onClick={(e) => console.log(e.target)}>
                  All
                </Dropdown.Item>
                <Dropdown.Item>Open</Dropdown.Item>
                <Dropdown.Item>Fullfilled</Dropdown.Item>
                <Dropdown.Item>Close</Dropdown.Item>
              </Dropdown.Menu>
            </Dropdown.Content>
          </Dropdown>
        </div>

        <div style={{ padding: '10px' }}>
          <Dropdown action="over" size={300}>
            <Dropdown.Trigger as="span">
              <Dropdown.Button>
                Dropdown <ChevronDownIcon width={12} height={12} />
              </Dropdown.Button>
            </Dropdown.Trigger>
            <Dropdown.Content>
              <Dropdown.Menu>
                <Dropdown.Item onClick={(e) => console.log(e.target)}>
                  All
                </Dropdown.Item>
                <Dropdown.Item>Open</Dropdown.Item>
                <Dropdown.Item>Fullfilled</Dropdown.Item>
                <Dropdown.Item>Close</Dropdown.Item>
              </Dropdown.Menu>
            </Dropdown.Content>
          </Dropdown>
        </div>
        <div style={{ padding: '10px' }}>
          <Table>
            <Table.Header>
              <Table.Row>
                <Table.HeaderCell>Header</Table.HeaderCell>
                <Table.HeaderCell>Header</Table.HeaderCell>
                <Table.HeaderCell>Header</Table.HeaderCell>
              </Table.Row>
            </Table.Header>

            <Table.Body>
              <Table.Row>
                <Table.Cell>Cell</Table.Cell>
                <Table.Cell>Cell</Table.Cell>
                <Table.Cell>Cell</Table.Cell>
              </Table.Row>
              <Table.Row>
                <Table.Cell>Cell</Table.Cell>
                <Table.Cell>Cell</Table.Cell>
                <Table.Cell>Cell</Table.Cell>
              </Table.Row>
              <Table.Row>
                <Table.Cell>Cell</Table.Cell>
                <Table.Cell>Cell</Table.Cell>
                <Table.Cell>Cell</Table.Cell>
              </Table.Row>
            </Table.Body>
          </Table>
        </div>
        <div style={{ padding: '10px' }}>
          <Table lined>
            <Table.Header>
              <Table.Row>
                <Table.HeaderCell colSpan={3}>Header</Table.HeaderCell>
              </Table.Row>
            </Table.Header>

            <Table.Body>
              <Table.Row>
                <Table.Cell rowSpan={2}>Cell</Table.Cell>
                <Table.Cell>Cell</Table.Cell>
                <Table.Cell>Cell</Table.Cell>
              </Table.Row>
              <Table.Row>
                <Table.Cell>Cell</Table.Cell>
                <Table.Cell>Cell</Table.Cell>
              </Table.Row>
              <Table.Row>
                <Table.Cell>Cell</Table.Cell>
                <Table.Cell>Cell</Table.Cell>
                <Table.Cell>Cell</Table.Cell>
              </Table.Row>
            </Table.Body>
          </Table>
        </div>
        <div style={{ padding: '10px' }}>
          <Toggle htmlFor="wadus" onChange={(state) => console.log(state)} />
          <Toggle htmlFor="foo" checked />
          <Toggle htmlFor="foone" disabled />
        </div>
        <div style={{ padding: '10px' }}>
          <Dropshadow>
            <div style={{ padding: '10px', background: '#fff' }}>Shadow 4</div>
          </Dropshadow>
        </div>
        <div style={{ padding: '10px' }}>
          <Dropshadow size={8}>
            <div style={{ padding: '10px', background: '#fff' }}>Shadow 8</div>
          </Dropshadow>
        </div>
        <div style={{ padding: '10px' }}>
          <Dropshadow size={16}>
            <div style={{ padding: '10px', background: '#fff' }}>Shadow 16</div>
          </Dropshadow>
        </div>
        <div style={{ padding: '10px' }}>
          <Range
            onChangeStart={(value) => console.log('start', value)}
            onChange={(value) => console.log(value)}
            onChangeComplete={(value) => console.log('complete', value)}
          />
        </div>
        <div style={{ padding: '10px' }}>
          <Range
            disabled
            width={300}
            maxValue={20}
            minValue={0}
            value={10}
            onChangeStart={(value) => console.log('start', value)}
            onChange={(value) => console.log(value)}
            onChangeComplete={(value) => console.log('complete', value)}
          />
        </div>
        <div style={{ padding: '10px' }}>
          <Range
            maxValue={30}
            minValue={0}
            value={{ min: 5, max: 10 }}
            onChangeStart={(value) => console.log('start', value)}
            onChange={(value) => console.log(value)}
            onChangeComplete={(value) => console.log('complete', value)}
          />
        </div>
        <div style={{ padding: '10px' }}>
          <Range
            draggable
            maxValue={30}
            minValue={0}
            value={{ min: 5, max: 10 }}
            onChangeStart={(value) => console.log('start', value)}
            onChange={(value) => console.log(value)}
            onChangeComplete={(value) => console.log('complete', value)}
          />
        </div>
        <div style={{ padding: '10px' }}>
          <Range
            maxValue={30}
            minValue={0}
            value={5}
            onChange={(value) => console.log(value)}
            step={0.5}
          />
        </div>
        <div style={{ padding: '10px' }}>
          <Breadcrumb>
            <Breadcrumb.Item path="/">Home</Breadcrumb.Item>
            <Breadcrumb.Item path="/foo">Foo</Breadcrumb.Item>
            <Breadcrumb.Item>Wadus</Breadcrumb.Item>
          </Breadcrumb>
        </div>
        <div style={{ padding: '10px' }}>
          <Loading />
        </div>
        <div style={{ padding: '10px', backgroundColor: '#000' }}>
          <Loading negative />
        </div>
        <div style={{ padding: '10px' }}>
          <Loading size={48} />
        </div>
        <div style={{ padding: '10px' }}>
          <Avatar url="" />

          <Avatar url="https://avatars0.githubusercontent.com/u/1366843" />
          <Avatar
            size={32}
            url="https://avatars0.githubusercontent.com/u/1366843"
          />
          <Avatar
            size={48}
            url="https://avatars0.githubusercontent.com/u/1366843"
          />
        </div>

        <div style={{ padding: '10px' }}>
          <Badge>Foo</Badge>

          <Badge color="rgba(128, 182, 34, 0.24)">Foo</Badge>
          <Badge as="span" color="skyblue">
            See you in Mars
          </Badge>
          <Badge color="#B4E0FA">Store 1</Badge>
          <Badge color="#E4D8EB" closable>
            Store 2
          </Badge>

          <Badge closable closeColor="#1785FB">
            Store 3
          </Badge>
        </div>

        <div style={{ padding: '20px 100px' }}>
          <Tooltip>
            <Tooltip.Content>Hola mundo</Tooltip.Content>
            <Tooltip.Trigger>tooltip</Tooltip.Trigger>
          </Tooltip>{' '}
          <Tooltip to="bottom">
            <Tooltip.Content>Hola mundo</Tooltip.Content>
            <Tooltip.Trigger>tooltip</Tooltip.Trigger>
          </Tooltip>
        </div>
        <div style={{ padding: '50px 100px' }}>
          <Tooltip to="left">
            <Tooltip.Content>Hola mundo</Tooltip.Content>
            <Tooltip.Trigger>tooltip</Tooltip.Trigger>
          </Tooltip>{' '}
          <Tooltip to="right">
            <Tooltip.Content>Hola mundo</Tooltip.Content>
            <Tooltip.Trigger>tooltip</Tooltip.Trigger>
          </Tooltip>
        </div>

        <div style={{ padding: '20px' }}>
          <div style={{ width: '300px', marginBottom: '20px' }}>
            <Collapsible>
              <Collapsible.Header>Header</Collapsible.Header>
              <Collapsible.Content>Content</Collapsible.Content>
            </Collapsible>
          </div>

          <div style={{ width: '300px' }}>
            <Collapsible open={false}>
              <Collapsible.Header>Header</Collapsible.Header>
              <Collapsible.Content>Content</Collapsible.Content>
            </Collapsible>
          </div>
        </div>

        <div style={{ padding: '20px' }}>
          <div style={{ width: '400px', marginBottom: '20px' }}>
            <Steps>
              <Steps.Header>Header</Steps.Header>
              <Steps.Content>Content 1</Steps.Content>
              <Steps.Content>Content 2</Steps.Content>
              <Steps.Content>Content 3</Steps.Content>
              <Steps.Content>Content 4</Steps.Content>
            </Steps>
          </div>

          <div style={{ width: '400px' }}>
            <Steps step={2}>
              <Steps.Header>Header</Steps.Header>
              <Steps.Content>Content 1</Steps.Content>
              <Steps.Content>Content 2</Steps.Content>
              <Steps.Content>Content 3</Steps.Content>
              <Steps.Content>Content 4</Steps.Content>
            </Steps>
          </div>
        </div>

        <div style={{ padding: '20px' }}>
          <Legend style={{ marginBottom: '20px' }}>
            <Legend.Panel>
              <Steps>
                <Steps.Header>Header</Steps.Header>
                <Steps.Content>Content 1</Steps.Content>
                <Steps.Content>Content 2</Steps.Content>
                <Steps.Content>Content 3</Steps.Content>
                <Steps.Content>Content 4</Steps.Content>
              </Steps>
            </Legend.Panel>
          </Legend>

          <Legend>
            <Legend.Panel>
              <Collapsible>
                <Collapsible.Header>Header</Collapsible.Header>
                <Collapsible.Content>Content</Collapsible.Content>
              </Collapsible>
            </Legend.Panel>
            <Legend.Panel>
              <Collapsible>
                <Collapsible.Header>Postal Code</Collapsible.Header>
                <Collapsible.Content>Content</Collapsible.Content>
              </Collapsible>
            </Legend.Panel>
          </Legend>
        </div>

        <div style={{ padding: '20px' }}>
          <Flag onClick={(e) => console.log('flag closed')}>
            <Flag.Icon>
              <AlertFillIcon color={colors.support02} width={12} height={12} />
            </Flag.Icon>
            <Flag.Content>
              <Text color={colors.type01} weight="medium">
                You are now conected
              </Text>
              <Text color={colors.type02}>
                You have been added to the group “New Store on this region”
              </Text>
            </Flag.Content>
          </Flag>
        </div>

        <div style={{ padding: '20px' }}>
          <Banner color="red">
            <Banner.Icon>
              <AlertFillIcon color={colors.white} width={12} height={12} />
            </Banner.Icon>
            <Banner.Content>
              <Text color={colors.type04}>
                An error has ocurred. Could not retreive customer information
              </Text>
            </Banner.Content>
          </Banner>
        </div>

        <div style={{ padding: '20px' }}>
          <Datepicker
            onDayClick={(range) => console.log(range)}
            initialMonth={new Date(2018, 9)}
            fromMonth={new Date(2018, 8)}
            toMonth={new Date(2020, 11)}
            disabledDays={[
              new Date(2018, 8, 12),
              new Date(2018, 8, 17),
              { after: new Date(2018, 8, 20), before: new Date(2018, 8, 25) }
            ]}
          />
        </div>

        <div style={{ padding: '20px' }}>
          <Datepicker
            fromMonth={new Date(2018, 8)}
            toMonth={new Date(2020, 11)}
            selectedDays={[
              { after: new Date(2018, 8, 20), before: new Date(2018, 8, 25) }
            ]}
          />
        </div>

        <div style={{ padding: '20px' }}>
          <Radiobutton.Group
            name="wadus"
            onChange={(state) => console.log(state)}
            selected="hola"
          >
            <Radiobutton value="hola">
              <Text>Hola</Text>
            </Radiobutton>
            <Radiobutton value="mundo">
              <Text>Mundo</Text>
            </Radiobutton>
          </Radiobutton.Group>

          <Radiobutton.Group name="foo" selected="hola" disabled>
            <Radiobutton value="hola">
              <Text>Hola</Text>
            </Radiobutton>
            <Radiobutton value="mundo">
              <Text>Mundo</Text>
            </Radiobutton>
          </Radiobutton.Group>
        </div>

        <div style={{ padding: '20px' }}>
          <Checkbox value={1} onChange={(state) => console.log(state)} />
          <Checkbox value="wadus" checked>
            <Text>Hola</Text>
          </Checkbox>

          <Checkbox value="wadus" checked disabled>
            <Text>Hola</Text>
          </Checkbox>
        </div>

        <div style={{ padding: '20px' }}>
          <Typeahead />
          <Typeahead query="wadus" />
          <Typeahead loading />
        </div>
      </div>
    );
  }
}

export default App;
